databricks_cli/cli.py

```diff
@@ -70,5 +74,119 @@ def cli():
 cli.add_command(repos_group, name='repos')
 cli.add_command(unity_catalog_group, name='unity-catalog')
 
+
+def _trampoline_into_new_cli():
+    trampoline_disable_env_var = 'DATABRICKS_CLI_DO_NOT_EXECUTE_NEWER_VERSION'
+    if os.environ.get(trampoline_disable_env_var) is not None:
+        return
+
+    # Try to trampoline only if we're running the CLI as 'databricks'.
+    if os.path.basename(sys.argv[0]) != 'databricks':
+        return
+
+    # Check to see if the new version of the CLI is in $PATH.
+    paths = os.environ['PATH'].split(os.pathsep)
+    self = None
+    self_version = version
+    candidate = None
+    for path in paths:
+        exec_path = os.path.join(path, 'databricks')
+        if os.name == 'nt':
+            exec_path = os.path.join(path, 'databricks.exe')
+
+        if not os.path.exists(exec_path):
+            continue
+
+        # Keep a pointer to the first 'databricks' in $PATH.
+        if not self:
+            self = exec_path
+
+        # The new CLI is a single binary larger than 1MB.
+        # We use this as heuristic to tell if the new CLI is installed.
+        # Because this version of the CLI is much smaller, we do not
+        # need to dedup our own path to avoid an exec loop.
+        stat = os.stat(exec_path)
+        if stat.st_size < (1024 * 1024):
+            continue
```

Comment:
I hope we don't break this...

If we're already loading the version of the CLI below, why don't we just call `version` anyways and get the result, breaking if we find something with version >= "0.100"?

Reply:
The version output in JSON only works on the new CLI, so if we remove this check we could be exec'ing into ourselves. The binary comes in at a decent 20MB so I think it's safe.

Are you thinking of another failure mode?

Reply:
Ah, I see... I was suggesting replacing this check with the version check. The other failure mode is if the python binary grows in size somehow.